src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,132 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [isValid, setIsValid] = useState({
+    imgUrl: true,
+    imdbUrl: true,
+  });
+
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+    description: '',
+  });
+
+  const isFieldsFilled = (movieToCheck: Movie): boolean => {
+    return Object.entries(movieToCheck).every(([key, value]) => {
+      return key !== 'description'
+        ? value.trim() !== ''
+        : typeof value === 'string';
+    });
+  };
+
+  const isUrlValid = (url: string): boolean => {
+    // eslint-disable-next-line max-len
+    const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+    return pattern.test(url);
+  };
+
+  const changeHandler = (name: string, value: string) => {
+    setNewMovie(currentMovie => ({
+      ...currentMovie,
+      [name]: value,
+    }));
+  };
+
+  const submitHandler = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const validImdbUrl = isUrlValid(newMovie.imdbUrl);
+    const validImgUrl = isUrlValid(newMovie.imgUrl);
+
+    if (validImdbUrl && validImgUrl) {
+      onAdd(newMovie);
+      setCount(currentCount => currentCount + 1);
+      setNewMovie({
+        title: '',
+        imgUrl: '',
+        imdbUrl: '',
+        imdbId: '',
+        description: '',
+      });
```

Comment:
better to make constant for initial form state and use in both places
const initialMovieState = {
   title: '',
   imgUrl: '',
   imdbUrl: '',
   imdbId: '',
   description: '',
}
```

```suggestion
      setNewMovie(initialMovieState);
```